src/basics.md

```diff
@@ -470,6 +471,84 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Processes a string of text using [`Regex::captures_iter`] to capture multiple
+phone numbers.  The example here is for US convention phone numbers.
+
+```rust, no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use regex::Regex;
+use std::fmt;
+
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+#
+#[derive (PartialEq, PartialOrd, Debug)]
+ enum CountryCode {
+     US,
+ }
+
+#[derive (PartialEq, PartialOrd, Debug)]
+struct PhoneNumber {
+    country: CountryCode,
+    number: String,
+}
```

Comment:
I have tried to do this, but I cannot figure out the lifetime.  Here's the problem:

```
error[E0597]: borrowed value does not live long enough
  --> src\main.rs:49:5
   |
47 |             number: &[area_code,cap[4].to_string(),cap[5].to_string()].concat()
   |                      ---------------------------------------------------------- temporary value created here
48 |         }
49 |     });
   |     ^ temporary value dropped here while still borrowed
...
57 | }
   | - temporary value needs to live until here
```

Here's the struct def:
```
#[derive (PartialEq, PartialOrd, Debug)]
struct PhoneNumber<'a> {
    number: &'a str,
}

// Allows printing phone numbers based on country convention.
impl<'a> fmt::Display for PhoneNumber<'a> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "1 ({}) {}-{}", &self.number[0..3], &self.number[3..6], &self.number[6..10])
    }
}
```
Can you advise?